code/game/machinery/stasis.dm

```diff
@@ -50,13 +80,51 @@
 	update_appearance()
 	return CLICK_ACTION_SUCCESS
 
+/// Toggle the filtering function on or off. This makes a bunch of scary-ass needles poke out of the bed.
+/// In exchange for having a passive function on hand to filter people's blood, it causes a little bit of damage.
+/// This damage only occurs upon initial activation, either through filtering starting by you being buckled or
+/// someone activating filtering on the bed you're buckled in already. Malicious doctors or silicons could toggle
+/// this over and over while you're helplessly buckled to the bed, though the effect of you being mutilated with
+/// needles will be obvious to onlookers.
+/obj/machinery/stasis/CtrlClick(mob/user)
+	if(!can_interact(user))
+		return
+	. = ..()
```

Comment:
```suggestion
	. = ..()
	if(!can_interact(user))
		return
```
Though `/atom/proc/CtrlClick()` already calls `can_interact` so this is wasteful but it doesn't matter that much